frontend/src/hooks/useActions.js

```diff
@@ -563,4 +504,39 @@ const promptLoadFile = (parse, onData, errorMessage = 'Failed to parse file') =>
   input.click()
 }
 
+// Takes in the IDs of states, comments, and transitions
+// Parameters also include  the current project and whether a template is being created
+// Outputs a CopyData to be copied or Template object to be created into a template
+const selectionToCopyTemplate = (stateIds, commentIds, transitionIds, project, isTemplate) => {
+  const selectedStates = stateIds.map((stateId) => {
+    return project.states.find((state) => {
+      return state.id === stateId
+    })
+  })
```

Comment:
Imo it'll be cleaner to use `filter` instead of `map` and `find`
```suggestion
  const selectedStates = project.states.filter(state => stateIds.includes(state))
```

Could also move it into a simple helper like (This is in typescript so its clearer, I understand the file is still JS)
```suggestion
  const filterFrom = <T>(allowed: T[], from: T[]): T[] => from.filter(x => allowed.includes(x))
  const selectedComments = filterFrom(commentIds, project.comments)
```
Either way is fine, but make sure to update the comments and transitions filtering also